detect_secrets/core/secrets_collection.py

```diff
@@ -363,11 +411,26 @@ def _extract_secrets_from_file(self, f, filename):
                 )
                 f.seek(0)
 
-        except UnicodeDecodeError as error:
-            log.warning(
-                '%s failed to load and could not be scanned.\nError: %s',
-                filename, str(error),
-            )
+        except UnicodeDecodeError:
+            if (is_binary(filename)):
+                # If the file type is binary, suppress the warning
+                pass
+            elif not suppress_unscannable_file_warnings:
+                log.warning(
+                    '%s failed to load, and could not be scanned,'
+                    ' because the file is not valid UTF-8.'
+                    '\nIf possible, convert this file to valid UTF-8 for it to be scanned.',
+                    filename,
+                )
+            if fail_on_file_unscannable:
```

Comment:
> With this if right here, when the file is not scannable because it is BINARY; it still fails the Build.....

Yeah, that `pass` I was using in the if statement to check if binary wasn't working as expected. It should now be fixed by this: https://github.com/IBM/detect-secrets/pull/121/commits/b356ce95468fdfbe4db0018443ea3a038d69354c

> Given that BINARY files are out of scope, shouldn't they be given a pass even if fail_on_file_unscannable=true? OR no (with why)?

Yeah they should be given a pass, the commit I linked above fixes that (I re-ran locally to verify that this is fixed).